packages/app/pages/course/[cid]/invite.tsx

```diff
@@ -1,77 +1,164 @@
-          </Head>
+      <div
+        style={{
+          display: "flex",
+          justifyContent: "center",
+          alignItems: "center",
+          height: "100vh",
+        }}
+      >
+        <Spin tip="Loading..." size="large" />
+      </div>
+    );
+  }
+
+  let cardElement;
 
-          <div
-            style={{
-              display: "flex",
-              justifyContent: "center",
-              alignItems: "center",
-              marginTop: 200,
-            }}
-          >
-            <Card
-              style={{ width: 600, textAlign: "center" }}
-              cover={
-                <img
-                  alt="example"
-                  height="200"
-                  style={{ objectFit: "cover" }}
-                  src="https://open-2021.sites.olt.ubc.ca/files/2020/10/OSIP-2020-Slider.jpg"
-                />
-              }
-            >
-              <Space
-                direction="vertical"
-                style={{ width: "100%" }}
-                size="large"
-              >
-                <Meta title={cardMetaTitle} description={cardMetaDescription} />
-                <Button
-                  type="primary"
-                  style={{ width: "100%", height: 50, borderRadius: "5px" }}
-                >
-                  Accept Invitation
-                </Button>
-              </Space>
-            </Card>
-          </div>
-        </StandardPageContainer>
-      </>
+  if (profile?.organization.id !== course?.organizationCourse.id) {
+    cardElement = renderCard(
+      "You cannot join a course that is not in your organization",
+      {},
+      "Back to my courses",
+      () => {
+        localStorage.removeItem("lastVisited");
+        router.push("/courses");
+      }
     );
-  } else if (course && code && code !== "demoCodeKey") {
-    return (
-      <>
-        <StandardPageContainer>
-          <Head>
-            <title>Page Not Found or is it?</title>
-          </Head>
-          page not found?
-        </StandardPageContainer>
-      </>
+  } else if (code !== course?.courseInviteCode) {
+    cardElement = renderCard(
+      "Invalid Course Code",
+      {},
+      "Back to my courses",
+      () => {
+        localStorage.removeItem("lastVisited");
+        router.push("/courses");
+      }
     );
   } else {
-    return <Spin tip="Loading..." size="large" />;
+    cardElement = renderCard(
+      `Invitation to join ‘${course?.name}‘`,
+      { title: cardMetaTitle, text: cardMetaDescription },
+      "Accept Invitation",
+      async () => {
+        const userData: UBCOuserParam = {
+          email: profile.email,
+          first_name: profile.firstName ?? "",
+          password: "",
+          last_name: profile.lastName ?? "",
+          selected_course: course.id,
+          sid: profile.sid,
+          photo_url: profile.photoURL,
+          courses: [],
+        };
+        await addStudent(userData);
+      },
+      <img
+        alt="example"
```

Comment:
`alt="Course image"`, we shouldn't keep it as example, it's used for accessibility reading